src/System.IO.FileSystem/tests/Enumeration/ErrorHandlingTests.netcoreapp.cs

```diff
@@ -16,27 +16,55 @@ public IgnoreErrors(string directory)
             { }
 
             public int ErrorCount { get; private set; }
+            public string DirectoryFinished { get; private set; }
 
             protected override string TransformEntry(ref FileSystemEntry entry)
-            {
-                throw new NotImplementedException();
-            }
+                => entry.FileName.ToString();
 
             protected override bool ContinueOnError(int error)
             {
                 ErrorCount++;
                 return true;
             }
+
+            protected override void OnDirectoryFinished(ReadOnlySpan<char> directory)
+                => DirectoryFinished = directory.ToString();
         }
 
         [Fact]
         public void OpenErrorDoesNotHappenAgainOnMoveNext()
         {
+            // What we're checking for here is that we don't try to enumerate when we
+            // couldn't even open the root directory (e.g. open the handle again, try
+            // to get data, etc.)
             using (IgnoreErrors ie = new IgnoreErrors(Path.GetRandomFileName()))
             {
                 Assert.Equal(1, ie.ErrorCount);
                 Assert.False(ie.MoveNext());
                 Assert.Equal(1, ie.ErrorCount);
+
+                // Since we didn't start, the directory shouldn't finish.
+                Assert.Null(ie.DirectoryFinished);
```

Comment:
Is it work doing another MoveNext, to make sure that doing a second one after a failed one doesn't throw?

Reply:
And below

Reply:
It wouldn't hurt, but not super likely to find an issue. I'm not worried enough about it to feed another change on this PR.